views-fieldset/src/main/java/io/micronaut/views/fields/fetchers/EnumRadioFetcher.java

```diff
@@ -0,0 +1,90 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.views.fields.fetchers;
+
+import io.micronaut.core.annotation.Experimental;
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.annotation.Nullable;
+import io.micronaut.core.util.StringUtils;
+import io.micronaut.views.fields.messages.Message;
+import io.micronaut.views.fields.elements.Radio;
+import jakarta.inject.Singleton;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.List;
+
+/**
+ * {@link RadioFetcher} implementation for Enums.
+ * @author Sergio del Amo
+ * @since 4.1.0
+ * @param <T> Field type.
+ */
+@Experimental
+@Singleton
+public class EnumRadioFetcher<T> implements RadioFetcher<T> {
+
+    private static final Logger LOG = LoggerFactory.getLogger(EnumRadioFetcher.class);
+
+    @Override
+    public List<Radio> generate(Class<T> type) {
+        if (type.isEnum()) {
+            return generateEnumRadioButtons((Class<? extends Enum>) type, null);
+        } else {
+            LOG.warn("Type {} is not an enum", type);
+        }
+        return Collections.emptyList();
+    }
+
+    @Override
+    public List<Radio> generate(T instance) {
+        if (instance.getClass().isEnum()) {
+            return generateEnumRadioButtons((Class<? extends Enum>) instance.getClass(), (Enum) instance);
+        } else {
+            LOG.warn("instance {} is not an enum", instance.getClass());
+        }
+        return Collections.emptyList();
+    }
+
+    @NonNull
+    private List<Radio> generateEnumRadioButtons(@NonNull Class<? extends Enum> type,
+                                                                  @Nullable Enum instance) {
+        return EnumSet.allOf(type).stream()
+            .map(it -> {
+                Radio.Builder builder = radioButtonFromEnum(type, (Enum) it);
+                if (instance != null && ((Enum<?>) it).name().equals(instance.name())) {
+                    builder.checked(true);
+                }
+                return builder.build();
+            })
+            .toList();
+    }
+
+    @NonNull
+    private Radio.Builder radioButtonFromEnum(@NonNull Class<? extends Enum> type, @NonNull Enum instance) {
+        String name = instance.name();
+        return Radio.builder().id(name.toLowerCase()).value(name).label(labelForBeanProperty(type, name));
+    }
+
+    @NonNull
+    private Message labelForBeanProperty(Class<? extends Enum> type, String name) {
+        String code = type.getSimpleName().toLowerCase() + "." + name.toLowerCase();
+        String defaultMessage = StringUtils.capitalize(name.toLowerCase().replaceAll("(.)([A-Z])", "$1 $2"));
```

Comment:
use a precompiled pattern. It seems like this logic is repeated again and again? Can it be moved to a static method or something

Reply:
resolved via https://github.com/micronaut-projects/micronaut-views/pull/613/commits/4181f07eba10cf92edd2de4e2eafb75a07346dd0